lib/Doctrine/ORM/Query/ResultSetMappingBuilder.php

```diff
@@ -432,19 +434,26 @@ public function generateSelectClause($tableAliases = [])
         foreach ($this->columnOwnerMap as $columnName => $dqlAlias) {
             $tableAlias = $tableAliases[$dqlAlias] ?? $dqlAlias;
 
-            if ($sql) {
+            if ($sql !== '') {
                 $sql .= ', ';
             }
 
-            $sql .= $tableAlias . '.';
-
             if (isset($this->fieldMappings[$columnName])) {
-                $class = $this->em->getClassMetadata($this->declaringClasses[$columnName]);
-                $sql  .= $class->fieldMappings[$this->fieldMappings[$columnName]]['columnName'];
+                $class             = $this->em->getClassMetadata($this->declaringClasses[$columnName]);
+                $fieldName         = $this->fieldMappings[$columnName];
+                $classFieldMapping = $class->fieldMappings[$fieldName];
+                $columnSql         = $tableAlias . '.' . $classFieldMapping['columnName'];
```

Comment:
@beberlei What we if i add a FieldMapping class with array access in another PR? How much will break? Or is there another better plan to get rid of some of the many arrays?

Reply:
that is a 3.0 task